crates/lox-derive/src/lib.rs

```diff
@@ -0,0 +1,724 @@
+        _ => None,
+    }
+}
+
+fn handle_version_field(
+    type_name: &proc_macro2::Ident,
+    field: &syn::Field,
+) -> Result<proc_macro2::TokenStream, proc_macro2::TokenStream> {
+    let string_type_name = type_name.to_string();
+
+    if !string_type_name.ends_with("Type") {
+        return Err(syn::Error::new_spanned(
+            type_name,
+            "Types with \"version\" field should be of the form SomethingType",
+        )
+        .into_compile_error());
+    }
+
+    let message_type_name = string_type_name
+        .trim_end_matches("Type")
+        .to_string()
+        .to_uppercase();
+
+    let field_name = field.ident.as_ref().unwrap();
+
+    // 7.4.4 Keywords must be uppercase and must not contain blanks
+    let expected_kvn_name = format!("CCSDS_{}_VERS", message_type_name);
+
+    // Unwrap is okay because we expect this span to come from the source code
+    let field_type = extract_type_path(&field.ty)
+        .unwrap()
+        .span()
+        .source_text()
+        .unwrap();
+    let field_type_new = extract_type_path(&field.ty).unwrap();
+
+    let parser = generate_call_to_deserializer_for_kvn_type(
+        &field_type,
+        field_type_new,
+        &expected_kvn_name,
+    )?;
+
+    Ok(quote! {
+        #field_name: #parser?,
+    })
+}
+
+fn deserializer_for_struct_with_named_fields(
+    type_name: &proc_macro2::Ident,
+    fields: &syn::punctuated::Punctuated<syn::Field, syn::token::Comma>,
+    is_value_unit_struct: bool,
+) -> proc_macro2::TokenStream {
+    if &type_name.to_string() == "UserDefinedType" {
+        //@TODO
+        return quote! {
+            Ok(Default::default())
+        };
+    }
+
+    if is_value_unit_struct {
+        let mut deserializer = None;
+        let mut unit_type: Option<String> = None;
+        let mut unit_field_name_ident: Option<&proc_macro2::Ident> = None;
+        let mut field_type: Option<String> = None;
+        let mut field_type_new: Option<&syn::Path> = None;
+
+        for (index, field) in fields.iter().enumerate() {
+            let field_name_ident = field.ident.as_ref().unwrap();
+
+            // Unwrap is okay because we only support named structs
+            let field_name = field_name_ident.span().source_text().unwrap();
+
+            match index {
+                0 => {
+                    if field_name.as_str() != "base" {
+                        return syn::Error::new_spanned(
+                            field,
+                            "The first field in a value unit struct should be called \"base\"",
+                        )
+                        .into_compile_error();
+                    }
+
+                    // Unwrap is okay because we expect this span to come from the source code
+                    let local_field_type = extract_type_path(&field.ty)
+                        .unwrap()
+                        .span()
+                        .source_text()
+                        .unwrap();
```

Comment:
In general, I'd prefer to use `expect` over `unwrap` throughout, because I've often been wrong about what's impossible and what isn't 😂

A little context makes life easy for whomever has to debug it, and helps document why not checking for the `None` case is fine to other devs at the call site (thanks for the comments, by the way).

Reply:
> I've often been wrong about what's impossible and what isn't 😂

I'm just built different.

Reply:
I will add this to TODO after initial merge, if that's okay with you.